src/main/java/com/minecolonies/coremod/colony/buildings/modules/AbstractAssignedCitizenModule.java

```diff
@@ -129,9 +129,7 @@ public final boolean hasAssignedCitizen()
     @Override
     public void serializeNBT(final CompoundTag compound)
     {
-        final CompoundTag assignedCompound = new CompoundTag();
-        assignedCompound.putInt(TAG_HIRING_MODE, this.hiringMode.ordinal());
-        compound.put(getModuleSerializationIdentifier(), assignedCompound);
+        compound.putInt(TAG_HIRING_MODE, this.hiringMode.ordinal());
```

Comment:
Did you test loading a larger old world with the new version?

Reply:
yes, I've used the competition world